app/src/main/java/io/neurolab/program_modes/MemoryGraphActivity.java

```diff
@@ -50,13 +63,24 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
                 WindowManager.LayoutParams.FLAG_FULLSCREEN);
         setTitle(R.string.mem_graph);
 
-        MainActivity mainActivity = new MainActivity();
-        rawData = mainActivity.getDeviceData();
-        if(rawData != null)
-        parseDataForGraph(rawData);
-
         memGraph = findViewById(R.id.mem_graph);
 
+        LayoutInflater layoutInflater = getLayoutInflater();
+        View progressView = layoutInflater.inflate(R.layout.progress_dialog_layout, null);
+        AlertDialog.Builder progress = new AlertDialog.Builder(this);
+        progress.setView(progressView);
+        progress.setCancelable(false);
+        progressDialog = progress.create();
+
+        permission = PermissionUtils.checkRuntimePermissions(this, READ_WRITE_PERMISSIONS);
+        initializeMemGraph(memGraph);
+    }
+
+    private void initializeMemGraph(LineChart memGraph) {
+
+        float yAxisMin = 5000f;
```

Comment:
There will be values below this limit. Let's not add a limit to y axis

Reply:
Okay @CloudyPadmal But then it will be difficult to view the graph on the Android screen for the users without limits. The user would need to scroll to various parts where the graph gets plotted. I had tried it. :sweat_smile: But anyways having said that, should i still remove the limits. Also Maybe we can have a function for finding the the max and min values for the y axis. Please let me know if we can do that.

Reply:
Okay will do that.

Reply:
Yes we need to set the limits dynamically.